Source/objects.cpp

```diff
@@ -1913,32 +1915,26 @@ void OperateBook(Player &player, Object &book, bool sendmsg)
 	}
 
 	if (setlevel && setlvlnum == SL_VILEBETRAYER) {
-		bool missileAdded = false;
-		for (int j = 0; j < ActiveObjectCount; j++) {
-			Object &questObject = Objects[ActiveObjects[j]];
-
-			Point target {};
-			bool doAddMissile = false;
+		Point target {};
+		if (book.position == Point { 26, 45 }) {
+			target = { 27, 29 };
+		} else if (book.position == Point { 45, 46 }) {
+			target = { 43, 29 };
+		} else {
+			return;
+		}
 
-			if (questObject._otype == OBJ_MCIRCLE2 && questObject._oVar6 == 1) {
-				target = { 27, 29 };
-				doAddMissile = true;
-			}
-			if (questObject._otype == OBJ_MCIRCLE2 && questObject._oVar6 == 2) {
-				target = { 43, 29 };
-				doAddMissile = true;
-			}
+		Object &circle = ObjectAtPosition(book.position + Direction::SouthWest);
+		assert(circle._otype == OBJ_MCIRCLE2);
 
-			if (doAddMissile) {
-				questObject._oVar6 = 4;
-				ObjectAtPosition({ 35, 36 })._oVar5++;
-				AddMissile(player.position.tile, target, Direction::South, MissileID::Phasing, TARGET_BOTH, player.getId(), 0, 0);
-				missileAdded = true;
-			}
-		}
-		if (!missileAdded) {
+		// Only verfiy that the player stands on the circle when it's the local player (sendmsg), cause for remote players the position could be desynced

```

Comment:
```suggestion
		// Only verfiy that the player stands on the circle when it's the local player (sendmsg), because for remote players the position could be desynced
```